controllers/storagecluster/external_ocs.go

```diff
@@ -0,0 +1,285 @@
+package storagecluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	externalClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	externalOCSResources = map[types.UID][]ExternalResource{}
+)
+
+// isExternalOCSProvider returns true if it is ocs to ocs ExternalStorage consumer cluster
+func isExternalOCSProvider(instance *ocsv1.StorageCluster) bool {
+	return instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS
+}
+
+// getExternalStorageRequestedCapacity returns requestedCapacity,
+// if not set it returns a default value 1Ti
+func getExternalStorageRequestedCapacity(instance *ocsv1.StorageCluster) *resource.Quantity {
+	if instance.Spec.ExternalStorage.RequestedCapacity == nil {
+		q := resource.MustParse("1Ti")
+		return &q
+	}
+	return instance.Spec.ExternalStorage.RequestedCapacity
+}
+
+// newProviderClient returns the *grpc.ClientConn and *providerClient.ConsumerClient
+func newProviderClient(instance *ocsv1.StorageCluster) (*grpc.ClientConn, *externalClient.OCSProviderClient, error) {
```

Comment:
Why do we need to return the grpc.ClientConn from this function?
We store it in the externalClusterClient, it should be enough to return it.

Reply:
This is required for closing the connection.